front/components/vaults/CreateVaultModal.tsx

```diff
@@ -0,0 +1,203 @@
+  handleMemberToggle: (member: string) => void
+): ColumnDef<RowData, unknown>[] {
+  return [
+    {
+      id: "name",
+      cell: (info: Info) => (
+        <DataTable.Cell avatarUrl={info.row.original.icon}>
+          {info.row.original.name}
+        </DataTable.Cell>
+      ),
+    },
+    {
+      id: "action",
+      cell: (info: Info) => {
+        const isSelected = selectedMembers.includes(info.row.original.userId);
+        return (
+          <Button
+            label={isSelected ? "Remove" : "Add"}
+            onClick={() => handleMemberToggle(info.row.original.userId)}
+            variant={isSelected ? "tertiary" : "secondary"}
+            size="sm"
+            icon={isSelected ? MinusIcon : PlusIcon}
+          />
+        );
+      },
+    },
+  ];
+}
+
+export function CreateVaultModal({
+  owner,
+  isOpen,
+  onClose,
+  onSave,
+  allUsers,
+}: CreateVaultModalProps) {
+  const [vaultName, setVaultName] = useState<string | null>(null);
+  const [selectedMembers, setSelectedMembers] = useState<string[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [isSaving, setIsSaving] = useState(false);
+
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const handleMemberToggle = useCallback((member: string) => {
+    setSelectedMembers((prevSelectedMembers) => {
+      const isCurrentlySelected = prevSelectedMembers.includes(member);
+      if (isCurrentlySelected) {
+        return prevSelectedMembers.filter((m) => m !== member);
+      } else {
+        return [...prevSelectedMembers, member];
+      }
+    });
+  }, []);
+
+  const createVault = async () => {
+    setIsSaving(true);
+    try {
+      const res = await fetch(`/api/w/${owner.sId}/vaults`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: vaultName,
+          members: selectedMembers,
+        }),
+      });
+
+      if (!res.ok) {
+        const errorData = await res.json();
+        throw new Error(errorData.error?.message || "Failed to create vault");
+      }
+      return await res.json();
+    } finally {
+      setIsSaving(false);
+    }
+  };
+
+  const rows = useMemo(() => getTableRows(allUsers), [allUsers]);
+
+  const columns = useMemo(
+    () => getTableColumns(selectedMembers, handleMemberToggle),
+    [selectedMembers, handleMemberToggle]
+  );
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      title="Create a Vault"
+      saveLabel="Create"
+      variant="side-md"
+      hasChanged={!!vaultName && selectedMembers.length > 0}
+      isSaving={isSaving}
+      onSave={async () => {
+        try {
+          await createVault();
+          sendNotification({
+            type: "success",
+            title: "Successfully created vault",
+            description: "Vault was successfully created.",
+          });
+        } catch (err) {
+          logger.error(
```

Comment:
Maybe we can catch the error "vault_already_exists" ?

Reply:
➕